src/NuoJsConnection.cpp

```diff
@@ -469,19 +469,29 @@ NuoDB::PreparedStatement* Connection::createStatement(std::string sql, Local<Arr
                     statement->setString(sqlIdx, toString(value).c_str());
                     break;
                 }
+                case NuoDB::NUOSQL_DATE: {
+                    char buffer[80];
+                    time_t seconds = (time_t)(toInt64(value) / 1000);
+                    struct tm* timeinfo;
+                    timeinfo = localtime(&seconds);
+                    strftime(buffer, 80, "%F %T", timeinfo);
+                    // buffer = "YYYY-MM-DD HH:MM:SS" -- 19 characters long
+                    int ms = toInt64(value) % 1000;
+                    buffer[19] = '.';
+                    buffer[20] = '0' + ms / 100;
+                    ms %= 100;
+                    buffer[21] = '0' + ms / 10;
+                    ms %= 10;
+                    buffer[22] = '0' + ms;
```

Comment:
This is a bit hacky, but I couldn't find another easy solution that didn't involve using non-standard packages or increasing our C standard.

Reply:
This looks brittle.

Is it correct to always append ".000" if the time is an even multiple of 1000?

Why is the value coming in a time_t?

Reply:
Was this discussed with the SQL team?  We should probably look at other database product mappings, or perhaps someone can state what requirements Bob was working under, maybe nobody was watching his mapping or gave it any particular thought.  However, we should get this right before the 3DMessage guys go into production so when they do there conversions, there is not chance we ask them to convert again after production.